pkg/orchestrator/etcd_worker.go

```diff
@@ -0,0 +1,260 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package orchestrator
+
+import (
+	"bytes"
+	"context"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/ticdc/pkg/etcd"
+	"go.etcd.io/etcd/clientv3"
+	"go.etcd.io/etcd/mvcc/mvccpb"
+	"go.uber.org/zap"
+)
+
+// EtcdWorker handles all interactions with Etcd
+type EtcdWorker struct {
+	client  *etcd.Client
+	reactor Reactor
+	state   ReactorState
+	// rawState is the local cache of the latest Etcd state.
+	rawState       map[string][]byte
+	pendingUpdates []*etcdUpdate
+	// revision is the Etcd revision of the latest event received from Etcd
+	// (which has not necessarily been applied to the ReactorState)
+	revision int64
+	prefix   string
```

Comment:
I have several problems
- How many etcd workers will be maintained, will they be classified by `prefix`
- What happens when an EtcdWorker manages a prefix, while some `reactors` only care about a sub-prefix, such as EtcdWorker manages `/tidb/cdc/job`, and each reactor only updates `/tidb/cdc/job/<reactor-id>`
- Will all etcd key-value operations be executed in serialize

Reply:
- Ideally each instance of owner or processor should maintain one `EtcdWorker`, and they should watch the prefix that is the highest level in the CDC key space.
- We can either maintain separate EtcdWorkers for each job, or create a super-reactor that dispatches events according to the subprefix.
- Yes, reactors see changes in the linearized order determined by the Etcd server. And reactors can see all changes it has made in previous ticks.